.gitlab/dev_container_deploy/docker_linux.yml

```diff
@@ -244,19 +244,30 @@ dev_nightly-dogstatsd:
     IMG_DESTINATIONS: dogstatsd-dev:nightly-${CI_COMMIT_REF_SLUG}-${CI_COMMIT_SHORT_SHA},dogstatsd-dev:nightly-${CI_COMMIT_REF_SLUG}
 
 # push images to `datadog-agent-qa` ECR for the end-to-end tests defined in `e2e.yml`
-qa_agent:
+qa_agent_windows:
   extends: .docker_publish_job_definition
   stage: dev_container_deploy
   rules:
     !reference [.on_container_or_e2e_changes_or_manual]
   needs:
-    - docker_build_agent7
-    - docker_build_agent7_arm64
     - docker_build_agent7_windows1809
     - docker_build_agent7_windows2022
   variables:
     IMG_REGISTRIES: agent-qa
-    IMG_SOURCES: ${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-amd64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-arm64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-win1809-amd64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-winltsc2022-amd64
+    IMG_SOURCES: ${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-win1809-amd64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-winltsc2022-amd64
+    IMG_DESTINATIONS: agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}
+
+qa_agent_linux:
+  extends: .docker_publish_job_definition
+  stage: dev_container_deploy
+  rules:
+    !reference [.on_container_or_e2e_changes_or_manual]
+  needs:
+    - docker_build_agent7
+    - docker_build_agent7_arm64
+  variables:
+    IMG_REGISTRIES: agent-qa
+    IMG_SOURCES: ${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-amd64,${SRC_AGENT}:v${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}-7-arm64
     IMG_DESTINATIONS: agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}
```

Comment:
I think that this won’t work:
`qa_agent_linux` and `qa_agent_windows` are two jobs that will try to generate a docker image with exactly the same name `agent:${CI_PIPELINE_ID}-${CI_COMMIT_SHORT_SHA}`.
So, depending on which job finishes the last, this image will either have only the Linux images or only the Windows ones.
But for the `containers` EKS tests, we need an image that contains everything.

Reply:
We could publish the three images, but it would add complexity to the code in test infra - we could have a `agent-linux` that has only linux. How much wait time does having both in one image takes, compared to the only linux ?

Reply:
It's a matter of 10 to 15 minutes - If it's not worth it I'll revert and keep a unified `qa_agent`!